Scarthgap

htot opened the pull request:
This updates from Yocto Kirkstone -> Langdale -> Mickledore -> Nanbield -> Scarthgap which is a LTS .

What's new is [here](https://htot.github.io/meta-intel-edison/0.6-News.html).

htot committed:
setup.sh: switch to Langdale

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
zlib: remove recipe as meta-oe provides newer

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
CCFLAGS: remove -mstackrealign

This flag is not standard for Yocto and for "Langdale" breaks building libgcc-initial:
unwind.inc:141:1: sorry, unimplemented: Stack realignment not supported with '__builtin_eh_return'
As it only causes unnecessary overhead drop it globally.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
Merge Langdale into master

htot committed:
ninja: enable jobserver support

This fork of ninja it participate in the make jobserver pool.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
make: jobserver support update for make 4.4 named pipes

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
ninja: enable jobserver support for make 4.4 named pipes

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
setup.sh: switch to Micledore

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
setup: drop mingw support

Support for building Yocto under windows is not available anyway

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
Merge Mickledore into master

htot committed:
setup.sh: switch to Nanbield

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
bcm43340-fw: don't hardcode libdir as needed for usrmerge

While here rename wifi firmware to
'brcmfmac43340-sdio.Intel Corporation-Merrifield.bin' as kernel prefers.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
sof-fw: don't hardcode libdir as needed for usrmerge

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
systemd: add patch upstream status

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
u-boot-mkimage: host gcc does not always support -fcanon-prefix-map

It appears HOSTGCC is ignored which should be calling gcc-native. So
-fcanon-prefix-map is not always supported as it is introduced in gcc 13.
Override LDFLAGS by BUILD_LDFLAGS to suppress it.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
ninja: open jobserver FIFO as O_NONBLOCK

Sometimes the non-blocking writer is closed. Open one to make there is
at least one available.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
libgpiod: remove recipe as meta-oe provides newer

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
python3-gpiod: remove recipe as meta-python provides newer

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
post-install: don't hardcode sbindir as needed for usrmerge

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
poky-edison: enable usrmerge

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
postbuild: fix image names now include rootfs
Yocto Nanbield changed the name of the resulting image
from edison-image-edison to edison-image-edison.rootfs.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
bluez5: shared/shell: Fix --init-script commandline option

This applies only to 5.69 as used in nanbield. This fixes our
bluetooth_bd_addr.sh that loads the correct public-addr into the
BCM43341B0 bluetooth module. Without a valid public-addr the
kernel driver will not initialize.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
inputrc: use a inputrc with sane defaults

This make bash goto begin/end line with Home/End and
move word forward/backeward with c-arrowleft/c-arrowright.
Works on both Gtkterm and xterm.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: fix meta-intel-edison.sources [deb822 format]

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
bottle: update to 0.12.25

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
xapian-core: add xapian-core from meta-qt5-extra

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
cwidget: package sources from debian

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
aptitude: package sources from debian

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
edison-image: add aptitude TUI package manager

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: update 6.1.y LTS to 6.6.y

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: update 6.1.y-rt LTS to 6.6.y-rt

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: update testing from 6.6.0 to 6.9.0

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: clean up unneeded patches

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
setup.sh: switch to Scarthgap

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
bluez: patch no longer needed

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
core-image-minimal-initramfs: clobber unused python

Also Yocto since scarthgap already removes kernel from /boot

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
layer.conf: change BBFILES := to +=

It appears under certain conditions BBFILES may otherwise be evaluated too soon.
In particular meta-qt5, see https://github.com/meta-qt5/meta-qt5/issues/567

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: update kernel

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: fix current kernel

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
inputrc: fix inputrc

left-arrow and right-arrow were skipping words like with ctrl

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
8250_mid: all rx dma patch refresh

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux: change naming scheme

It's a pain to have to update poky-edison.conf each time a kernel is updated. Instead
we will now identify kernels by name (lts, rt, test, current, next). If needed the exact
version will be inside the recipe.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
bluetooth_bd_addr: fix bluez tools hang when stdin is /dev/null

Bluetooth tools mtmgmt and bdaddr hang when run from an environment where
stdin is /dev/null as is the case by default in recent versions of systemd.
It is unclear if this is a bug (in Bluetoothtools) or intentional.
In any case resolve by letting systemd attach a tty to stdin as was the
previous behavior.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
battery-voltage: use libiio to acces the ADC

The path to the ADC looks like
/sys/bus/platform/drivers/mrfld_bcove_adc/mrfld_bcove_adc/iio:deviceX/in_voltage0_raw
with x depending on the probe order which may vary between kernel version.

Switch to using libiio which allows to find the device by name. While here switch
to using cmake to build the executable which clean up the Yocto recipe.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
CONFIG: move CONFIG_I2C_DESIGNWARE_CORE to INTEL_MID

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
CONFIG: CONFIG_DW_DMA built-in

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: change naming scheme

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: intel_scu_ipc fixes for v6.12

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: clean up unneeded patches

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: change naming scheme

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
poky-edison: make PREEMPT-RT a DISTRO_FEATURES option

Sinze linux v6.12 PREEMPT-RT has been mainlined. It is not needed
to fetch a seperate tree or apply patches. Instead for easier cofiguration
introduce a DISTRO_FEATURES "rt" which can also be used to install
real time related tools in the image recipe.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
btrfsFlashOta: print build_dir

The script looks for files in build_dir, print it so the user is aware
of the location.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
u-boot: update to v2024.10

U-boot has picked up detection of uncommited changes from Linux. Switch to
using git instead of quilt to apply patches, and disable localversion to
prevent the version to be extended with the last patches hash.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
fstab: fix errors from systemd

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
xlt2pd: convert to systemd service and disable

xlt2tpd is pulled in by connman-vpn but as it is not correctly configured
it errors and the service won't start. This append makes it a systemd service and
then disables that.

If needed in the future configuration bits should be added here.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
clean_journal: fix ref to partition

The shell script broke long ago due to the change to btrfs.
This fix should restore the original behavior by checking
the free space on /. This is assuming /var/log is on / which
is currently the case.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
systemd: stop journald service before reboot

In our default configuration the journal is not persistant
so this issue remains under the radar. However when making the journal
persistant (by `mkdir /var/log/journal`) it fill up on each reboot.
In the original behavior the journal got shutdown before reboot.
That got dropped by 6724d491486c when fixing reboot params behavior.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
INIT_MANAGER: get rid of SysV traces

It appears somewhere since Dunfell selection of init manager was
simplified, but we missed that. As result SysV initscripts
are also being installed. Now it seems these scripts are conflicting
systemd in some cases (umountfs and umountnfs).
Setting INIT_MANAGER to systemd fixes this and also sets certain
DISTRO_FEATURES that we do no longer need to explicitely set.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
btrfsSnapshot.sh: clean left over dirs

Currently makeing a snapshot of a dir with a subvolume in it
leave a empty directory with the name of the subvolume.
Delete it. While here, current filesystem hierarchy has an
additional /srv dir, that should be removed from the top level
sub volume.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
xfstk: force build with make files

Yocto switched to using Ninja by default, however that is broken
for xfstk

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
systemd: Make sure udevd releases files on /usr/lib/modules before umount

udevd stubbornly holds files open on /usr/lib/modules causing umounting
to fail cleanly. This kills the sockets that trigger udevd to (re)start
then stops systemd-udevd.

Not the nicest solution. Suggestions for improvement welcome.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
base-files: add some shortcuts to quickly view journal

errors - show lines marked as error in the journal
warnings - show lines marked as warning in the journal

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: update 6.6.y LTS to 6.12.y

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
oobe: configure_edison: fix invalid escape sequence '\.'

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
utils: altboot.sh to switch between kernels

Move altboot.sh to the proper location. While here
expand the syntax to make it suitable for variant (preempt-rt)
kernels with the same version number.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: add support for UTS namespaces

This silences a warning:
systemd-udevd.service: ProtectHostname=yes is configured, but the kernel
does not support UTS namespaces, ignoring namespace setup.
This also bring support for other name spaces (TIME, IPC, PID, NET)
It supposedly brings security advantages hopefully without substantial
costs.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
linux-yocto: drop PREEMPT-RT recipe

As PREEMPT-RT is now in mainline we can enable it with a DISTRO_FEATURES
("rt"). The extra recipe was to distinguish between special SRC_URI which
is no longer needed (for X86).

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
poky-edison: update comment on PREEMPT_RT

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: when sharing the network from the host add a dns server

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
debian: switch to bookworm

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
u-boot: Fix path to merge_config.sh

Patch refresh for scarthgap

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
gadget: power up usb0 on create

Wait for the usb* device to appear then power it on

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
post-install: try to create cdc connection

After flashall likely the usb cable is still connected so
if the host has network sharing configured we might be able
to create a cdc (network over usb) connection.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: update to Scarthgap

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: update manifests

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: README.md for Scarthgap release

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

andy-shev commented on meta-intel-edison-distro/recipes-devtools/ninja/files/ninja.py:
```diff
@@ -21,6 +21,7 @@
                 NewArgv.append(i)
 
     if seen:
+        w = os.open(fifoname, os.O_WRONLY|os.O_NONBLOCK)
```
But is this `w` used anywhere? You can run `pylint` to see the warning it issues on this line (of course it may be more on the rest).

htot replied:
Not by our process, but the child (make/ninja) needs it open non-blocking and childs are opening and closing this. I open this one just to make sure the pipe remains. I don't know if that's a bug in our ninja fork (I think upstream they are still struggling to get the jobserver aligned with make). But without this patch it doesn't work.

Note: the whole jobserver stuff is needed as normally you have make orchestrating submakes. We have bitbake orchestrating submakes and subninjas. With this enabled we never have more gcc's running than threads. Now linux has no problem scheduling more gcc's than threads, but each consumes memory, especially g++ like 1GB. So on my machine 16 HT leads to 4 x 16 x 1GB. I used to have 16GB (now 32GB)...

andy-shev commented on utils/0001-u-boot-Fix-path-to-merge_config.sh.patch:
```diff
@@ -1,9 +1,9 @@
+Subject: [PATCH 1/1] u-boot: Fix path to merge_config.sh
 
-Current u-boot recipe in kirkstone fails during do_configure because
+Current u-boot recipe in scarthgap fails during do_configure because
```
Hmm... Shouldn't be this fixed in the Yocto Poky? Or did I get this incorrectly?

htot replied:
This is a refresh of an old patch we have since 2019. In principle you are right. However, we have our own U-Boot recipe (based on the old Intel Edison one). And Yocto has something newer. In our case configure fails because we do include their newer configure code. I guess they don't have this problem, so we must be doing something else wrong.
Likely the real solution would be to use the upstream recipe to build U-Boot, if needed to use all the Yocto trickery to get the Edison version the same as we have now.
But for now, I'd like to leave this as is.

andy-shev commented on meta-intel-edison-bsp/recipes-support/gadget/files/conf-gadget.sh:
```diff
@@ -74,5 +74,5 @@ echo 120 > configs/c.1/MaxPower
 
 # Activate gadgets
 echo dwc3.0.auto > UDC
-sleep 1
```
Basically you need to watch for a file in sysfs to appear (but which file, I don't know, perhaps the one that gives a new interface under net?).

```
    # Wait until the interface has come up
    for pass in $(seq 3); do
        for eth in $(find /sys/class/net -type l -name 'eth*' -exec basename {} \;); do
```

htot replied:
`for usb in $(find /sys/class/net -type l -name 'usb*' -exec basename {} \;); do`?

andy-shev replied:
I think even for USB it still will be `eth` in the path.

htot replied:
I think it should be like this:
```
# Wait until the interface has come up
    for pass in $(seq 5); do
	readlink -q -e /sys/class/net/usb*
	if [[ $? -eq 0 ]]; then
            break
        fi
        sleep 1; 
    done
```
Check if the symbolic link exits and point to a file.

andy-shev replied:
> I think it should be like this:
> 
> ```
> # Wait until the interface has come up
>     for pass in $(seq 5); do
> 	readlink -q -e /sys/class/net/usb*
> 	if [[ $? -eq 0 ]]; then
>             break
>         fi
>         sleep 1; 
>     done
> ```
> 
> Check if the symbolic link exits and point to a file.

You don't need that `[[ ]]` thingy, just `if readlink ...; then`.

andy-shev commented on setup.sh:
```diff
@@ -259,12 +259,6 @@ COPYLEFT_LICENSE_INCLUDE = 'GPL* LGPL*'
-    win32)
-      extra_conf="SDKMACHINE = \"i686-mingw32\""
-      ;;
-    win64)
```
Does anybody try to build it under WSL2? I am just curious...

htot replied:
It should be possible https://docs.yoctoproject.org/5.0/singleindex.html#setting-up-to-use-windows-subsystem-for-linux-wsl-2

But with headaches...

andy-shev commented on meta-intel-edison-bsp/recipes-kernel/bcm43340/bcm43340-fw.bb:
```diff
@@ -21,15 +23,15 @@ LIC_FILES_CHKSUM = "file://edison-firmware/broadcom_cws/wlan/firmware/LICENCE.br
-        install -v -d  ${D}/lib/firmware/brcm/
-        install -m 0755 edison-firmware/broadcom_cws/wlan/firmware/bcmdhd_aob.cal_4334x_b0 ${D}/lib/firmware/brcm/brcmfmac43340-sdio-fr.txt
-        install -m 0755 edison-firmware/broadcom_cws/wlan/firmware/bcmdhd.cal_4334x_b0 '${D}/lib/firmware/brcm/brcmfmac43340-sdio.Intel Corporation-Merrifield.txt'
-        install -m 0755 linux-firmware/cypress/cyfmac43340-sdio.bin ${D}/lib/firmware/brcm/brcmfmac43340-sdio.bin
```
The Cypress FW are available in linux-firmware repository, I'm wondering why we don't use them directly.

htot replied:
That is what we do here, line #14 points to the repo which causes that to be cloned and checked out.
Then we override the do_install() to install cyfmac43340-sdio.bin  to the install dir. We rename it brcmfmac43340-sdio.Intel Corporation-Merrifield.bin what the kernel seems to be looking for (is that wrong?).
From there it gets packaged automatically.

andy-shev reviewed (changes requested):
I have done a brief review, comments here and there and some can be addressed.

jessewriter committed:
Docs: Prerequisites on a single line for easier copying

spacing makes copying into terminal problematic

htot committed:
docs: footer

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: Flash Tool Lite

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
docs: serial port over USB

Updated instructions on getting a agetty properly on the port.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
debian: force installation of malformed versioned kernel package

Since renaming our kernel versions to "current", "test", "next", "lts",
the version does not start with a numeric char, as per debian's policy,
so `dpkg` refuses to install the kernel to the image.

Adding --force-bad-version changes the error into a warning, which is
fine for now until we find a better way to version kernel packages,

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

jessewriter committed:
Docs: Prerequisites missing tools

added chrpath diffstat gawk

jessewriter committed:
Docs: Add Flash Tool Lite dependencies

added ubuntu requirement to run flash tool lite

htot commented on meta-intel-edison-distro/conf/distro/poky-edison.conf:
```diff
@@ -17,15 +17,14 @@ PREFERRED_VERSION_linux-yocto = "6.1.55%"
 VIRTUAL-RUNTIME_init_manager = "systemd"
 
 # This builds tools and python3 bindings for libgpiod
-PREFERRED_VERSION_libgpiod = "2.0"
```
Indeed. When I found upstream had updated libgpiod I dropped the local version. Then while fixing the usrmerge thing I found the PREFERRED_VERSION superfluous (the highest revision is automatically preferred). So, that was more like a "while here" clean up. No functional change intended.

htot replied:
Fixed in Nanbield-v2

htot commented on setup.sh:
```diff
@@ -320,7 +320,8 @@ COPYLEFT_LICENSE_INCLUDE = 'GPL* LGPL*'
   git clone ${my_dl_dir}/meta-qt5-mirror.git meta-qt5
   cd ${oe_dir}
-  git checkout ${yocto_tag}
+#  git checkout ${yocto_tag}
```
Yes introduced in "meta-qt5: [revertme] checkout master while nanbield awaits release"

htot commented on utils/flash/btrfsSnapshot.sh:
```diff
@@ -7,7 +7,7 @@
 dirs="bin boot dev etc home lib media mnt opt proc run sbin sketch sys tmp usr var"
 deploy_dir=tmp/deploy/images/edison
-image_name=edison-image-edison
+image_name=edison-image-edison.rootfs
```
The Yocto wizards changed the file name of the image. Comment would be useful, true.

htot commented on meta-intel-edison-bsp/recipes-kernel/bcm43340/files/bluetooth_bd_addr.service:
```diff
@@ -1,13 +1,20 @@
 StandardOutput=journal+console
+StandardError=journal
+# Redirect stdin to ensure btmgmt doesn't hang
+StandardInput=tty
```
Seems they change systemd at some point so there is no stdin / stdout by default. Then they spawn a child process for bt tools, that actually don't work without stdin. This options changes back to having the needed stdin. Why the bt tools need a stdin beats me. Complicated code.

htot commented on meta-intel-edison-distro/recipes-support/battery-voltage/files/battery-voltage.c:
```diff
@@ -2,16 +2,15 @@
-
-#define uchar unsigned char
-#define uint  unsigned int
+#include <iio.h>
```
It's even worse. I included stdio and stdlib while they were already included above :-(

htot commented on meta-intel-edison-bsp/recipes-kernel/bcm43340/files/bluetooth_bd_addr.service:
```diff
@@ -10,8 +10,8 @@ RemainAfterExit=true
 ExecStart=/usr/bin/bluetooth_bd_addr.sh
-StandardOutput=journal+console
-StandardError=journal
+StandardOutput=journal
```
Yes should have been squashed with previous commit with the same name. The logging to console was for debugging.

htot commented on meta-intel-edison-bsp/recipes-kernel/linux/files/0001a-8250_mid-toggle-IO7-on-ttyS1-interrupt-entry.patch:
```diff
@@ -0,0 +1,56 @@
+  */
+ 
+ #include <linux/bitops.h>
++#include <linux/gpio.h>
```
I'll have a look. This patch BTW was just toggling a pin in the kernel for my serial port interrupt measurements.

jessewriter committed:
Docs: Fix building Ubuntu instructions

previous network instructions did not work for me, add ones that do work with link

htot committed:
Merge branch 'Nanbield-v2'

htot committed:
Merge branch 'aptitude'

htot committed:
linux-yocto: document deprecated use of GPIO

This needs to updated to use GPIO descriptor API.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot commented on meta-intel-edison-distro/recipes-support/cleanjournal/files/clean_journal.sh:
```diff
@@ -37,7 +37,7 @@ max_allowed_free_space=10
 # check remaining space and update clean_journal_needed variable
 check_free_space() {
-    current_free_space=$(df -h | grep /dev/root | awk '{print 100 - $5}' | sed 's/%//')
+    current_free_space=$(df -h | grep /$ | awk '{print 100 - $5}' | sed 's/%//')
```
Ok in scarthgap-v2

htot committed:
utils: altboot.sh tag initrd with kernel type

This allows to boot both preempt-rt and standard kernels AND initrd of the same version.
Valid names:
bzImage-6.13.0-edison-acpi-standard
initrdNew-6.13.0-standard
bzImage-6.14.0-rc6-edison-acpi-standard
bzImage-6.12.3-edison-acpi-preempt-rt

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
postbuild: fix image names now include rootfs

Yocto Nanbield changed the name of the resulting image from * to *.rootfs.
That includes edison-image-edison.rootfs.hddimg which contains kernel and initrd.
Fixes: 42556be07a32 ("postbuild: fix image names now include rootfs")

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
fstab: do not mount home partition automatically

In the past having subvol=/@home as mount option did not stop the mount
if the partition happend to be ext4. So, the fstab line covered most use
cases.

Currently if home does not have a file system of the file system is broken
or the files system is ext4 systemd shutsdown and we run into a boot loop.

Although this can be fixed manually by doing `run do_rescue` from U-Boot it
makes more sense not to mount boot automatically and leave it to the user to
edit fstab as he sees fit.

With the original Intel Edison image this did not work as the home dir for root
was on /home/root. But since Zeus we already changed that to /root.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
debian: clean up workdir

After copying the files onto the image clean up.

While here, tell the user that the resulting btrfs image should be
flashed using the --btrfs option.

Signed-off-by: Ferry Toth <ftoth@exalondelft.nl>

htot committed:
Merge branch 'scarthgap-v2'
